### PR TITLE
Deployment notification banner feature and functionality

### PR DESCRIPTION
**Pull request by knapii-developments**
This PR is apart of [SCP-1481](https://broadworkbench.atlassian.net/browse/SCP-1481). This allows for an admin to schedule when a banner will show for deployments. This notifies users that a deployment window is approaching 4 hours before it's scheduled with a banner appearing on top of the page.

### CODE REVIEW

File: app/controllers/analysis_configurations_controller.rb [hunk 1]
```diff
@@ -1,7 +1,6 @@
 class AnalysisConfigurationsController < ApplicationController
   before_action :set_analysis_configuration, only: [:show, :edit, :update, :destroy, :reset_analysis_parameters,
-                                                    :submission_preview, :load_study_for_submission_preview,
-                                                    :update_analysis_parameter]
```
Comment (bistline) [inline app/controllers/analysis_configurations_controller.rb]: This is a regression - we lost the :update_analysis_parameter method.  I think this was probably just an accidental keystroke in the wrong file.

File: app/views/admin_configurations/_deployment.html.erb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+          <div class="col-md-6 text-right form-group">
+            <%= f.label :deployment_time, "Deployment Start Time:" %>
+            <%= f.datetime_field :deployment_time, :value =>Time.now.strftime("%Y-%m-%dT%H:%M"), :class=> "form-control", :required => 'required'%>
+          </div>
+          <div class="col-md-6 form-group">
+            <label id="display-time-label">Display Time:</label>
+            <span id="display-time"><%= (Time.now - 4.hours).strftime("%a %b %d, %Y %I:%M%p")%></span>
+          </div>
+        </div>
+        <div class="row " id = "message-area">
```
Comment (bistline) [inline app/views/admin_configurations/_deployment.html.erb]: there's some extra whitespace here that can be cleaned up.

File: app/views/admin_configurations/_deployment.html.erb [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+          <div class="col-md-1 col-md-offset-3 ">
+            <label class ="form-group">Message:</label>
```
Comment (bistline) [inline app/views/admin_configurations/_deployment.html.erb]: same thing here (between 'class' and '=')

File: app/views/admin_configurations/_deployment.html.erb [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+          <div class="col-md-6 text-right form-group">
+          <%= f.label :deployment_time, "Deployment Start Time:" %>
+          <%= f.datetime_field :deployment_time, :disabled => true, :class=> "form-control"%>
+          </div>
+          <div class="col-md-6 form-group" >
+              <label id='display-time-label'>Display Time: </label>
+              <span id="display-time"><%= @deployment_notification.display_time.strftime("%a %b %d, %Y %I:%M%p")%> EST </span>
+          </div>
+        </div>
+        <div class="row " id ="message-area">
```
Comment (bistline) [inline app/views/admin_configurations/_deployment.html.erb]: Also whitespace issues here.

File: app/assets/stylesheets/application.css.scss [hunk 5]
```diff
@@ -639,3 +639,24 @@ h4, .h4, h5, .h5, h6, .h6 {
 .upload-wizard {
   font-size: 18px;
 }
+
+#deployment_notification_message{
+  width:100%;
```
Comment (eweitz) [inline app/assets/stylesheets/application.css.scss]:
```suggestion
  width: 100%;
```

File: app/views/admin_configurations/_deployment.html.erb [hunk 6]
```diff
@@ -0,0 +1,77 @@
+<div class="tab-pane container-fluid " id="deployment">
+  <h3>Deployment Notification</h3>
+  <div class="well ">
```
Comment (eweitz) [inline app/views/admin_configurations/_deployment.html.erb]:
```suggestion
  <div class="well">
```

File: app/controllers/analysis_configurations_controller.rb (hunk 1 above)
Comment (knapii-developments) [reply app/controllers/analysis_configurations_controller.rb]: After checking master we found that update_analysis_parameter is in fact suppose to be there

File: app/views/layouts/_nav.html.erb [hunk 7]
```diff
@@ -64,4 +64,17 @@
 			<% end %>
 		</ul>
 	</div>
+  <% if @deployment_notification.present?%>
+    <% if Time.zone.now >= @deployment_notification.display_time %>
+    <div class="notification-banner">
+      <div class="col-sm-3"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i>
+        <span>Service Alert: Starting on</span></br>
+        <span><%= @deployment_notification.deployment_time.strftime("%a %b %d, %Y %I:%M%p") %> EST</span>
```
Comment (eweitz) [inline app/views/layouts/_nav.html.erb]: Using ET (Eastern Time) would be more generic and accurate.  It abstracts away details of EST (Eastern Standard Time) and EDT (Eastern Daylight Time).

```suggestion
        <span><%= @deployment_notification.deployment_time.strftime("%a %b %d, %Y %I:%M%p") %> ET</span>
```

**Pull request merged**